kernel/arch/i386/boot/bootparse.cpp

```diff
@@ -116,10 +123,166 @@ void px_parse_multiboot2(void *info)
             break;
         }
         default:
-            px_rs232_printf("Unknown Multiboot2 Tag: %d\n", tag->type);
+            px_rs232_printf("Unknown Multiboot2 tag: %d\n", tag->type);
             break;
         }
         // move to the next tag, aligning if necessary
         tag = (struct multiboot_tag*)(((uintptr_t)tag + tag->size + 7) & ~((uintptr_t)7));
     }
 }
+
+/*
+ *  ___ _   _          _     ___
+ * / __| |_(_)_ ____ _| |___|_  )
+ * \__ \  _| \ V / _` | / -_)/ /
+ * |___/\__|_|\_/\__,_|_\___/___|
+ */
+
+static void print_stivale2_mmap(struct stivale2_struct_tag_memmap* mmap)
+{
+    static const char *mmap_types[] = {
+        [0] = "Invalid",
+        [STIVALE2_MMAP_USABLE] = "Available",
+        [STIVALE2_MMAP_RESERVED] = "Reserved",
+        [STIVALE2_MMAP_ACPI_RECLAIMABLE] = "ACPI reclaimable",
+        [STIVALE2_MMAP_ACPI_NVS] = "Non-volatile storage",
+        [STIVALE2_MMAP_BAD_MEMORY] = "Bad RAM"
+    };
+    // List of memory map entries
+    struct stivale2_mmap_entry* mmap_list = (struct stivale2_mmap_entry*)mmap->memmap;
```

Comment:
Be a bit easier to read using auto on the left hand side

Reply:
Resolved in ced2fd7a2b501b80624e9e68ebf765e4d6644c52